test/unit/reporters/apiary-reporter-test.coffee

```diff
@@ -1,211 +1,516 @@
+      call = null
+      runId = '507f1f77bcf86cd799439011'
+      requestBody = null
+      beforeEach () ->
+        uri = '/apis/public/tests/runs'
+        reportUrl = "https://absolutely.fency.url/wich-can-change/some/id"
+
+        # this is a hack how to get access to the performed request from nock
+        # nock isn't able to provide it
+        getBody = (body) ->
+          requestBody = body
+          return body
+
+        call = nock(process.env['APIARY_API_URL']).
+          filteringRequestBody(getBody).
+          post(uri).
+          reply(201, JSON.stringify({"_id": runId, "reportUrl": reportUrl}))
+
+      it 'should set uuid', (done) ->
+        emitter = new EventEmitter
+        apiaryReporter = new ApiaryReporter emitter, {}, {}
+        emitter.emit 'start', blueprintData, () ->
+          assert.isNotNull apiaryReporter.uuid
+          done()
+
+      it 'should set start time', (done) ->
+        emitter = new EventEmitter
+        apiaryReporter = new ApiaryReporter emitter, {}, {}
+        emitter.emit 'start', blueprintData, () ->
+          assert.isNotNull apiaryReporter.startedAt
+          done()
+
+      it 'should call "create new test run" HTTP resource', (done ) ->
+        emitter = new EventEmitter
+        apiaryReporter = new ApiaryReporter emitter, {}, {}
+        emitter.emit 'start', blueprintData, () ->
+          assert.isTrue call.isDone()
+          done()
+
+      it 'should attach test run ID back to the reporter as remoteId', (done) ->
+        emitter = new EventEmitter
+        apiaryReporter = new ApiaryReporter emitter, {}, {}
+        emitter.emit 'start', blueprintData, () ->
+          assert.isNotNull apiaryReporter.remoteId
+          done()
+
+      it 'should attach test run reportUrl to the reporter as reportUrl', (done) ->
+        emitter = new EventEmitter
+        apiaryReporter = new ApiaryReporter emitter, {}, {}
+        emitter.emit 'start', blueprintData, () ->
+          assert.isNotNull apiaryReporter.reportUrl
+          done()
+
+      it 'should have blueprints key in the request and it should be an array and members should have propers structure', (done) ->
+        emitter = new EventEmitter
+        apiaryReporter = new ApiaryReporter emitter, {}, {}
+        emitter.emit 'start', blueprintData, () ->
+          parsedBody = JSON.parse requestBody
+          assert.isArray parsedBody.blueprints
+          for blueprint in parsedBody.blueprints
+            assert.property blueprint, 'raw'
+            assert.property blueprint, 'file'
+            assert.property blueprint, 'parsed'
+          done()
+
+    describe 'when adding passing test', () ->
+      call = null
+      runId = '507f1f77bcf86cd799439011'
+      test = null
+      requestBody = null
+
+      beforeEach () ->
+        uri = '/apis/public/tests/steps?testRunId=' + runId
+
+        # this is a hack how to get access to the performed request from nock
+        # nock isn't able to provide it
```

Comment:
This is doing something completely different. I'm not matching mocks based on request body, I'm mocking all requests with any body in that url, but I'm setting expectations on the body itself, regardless it will be mocked or it will be not.